src/turbolinks/history.coffee

```diff
@@ -20,16 +23,27 @@ class Turbolinks.History
 
   push: (location, restorationIdentifier) ->
     location = Turbolinks.Location.wrap(location)
+    locationStack.push(location)
     @update("push", location, restorationIdentifier)
 
   replace: (location, restorationIdentifier) ->
     location = Turbolinks.Location.wrap(location)
     @update("replace", location, restorationIdentifier)
 
+  back: (location, restorationIdentifier) ->
```

Comment:
Actually, given the actual behavior of this PR, and your comment that "The expected behavior would be that if the previous url in the history is the url provided", perhaps `refresh` is a better name.